lib/ai/assistant.ts

```diff
@@ -1,13 +1,250 @@
+            required: ["id"]
+          }
+        }
+      },
+      {
+        type: "function",
+        function: {
+          name: "updateTask",
+          description: "Update an existing task",
+          parameters: {
+            type: "object",
+            properties: {
+              id: {
+                type: "number",
+                description: "The ID of the task to update."
+              }
+            },
+            required: ["id"]
+          }
+        }
+      },
+      {
+        type: "function",
+        function: {
+          name: "deleteTask",
+          description: "Delete an existing task",
+          parameters: {
+            type: "object",
+            properties: {
+              id: {
+                type: "number",
+                description: "The ID of the task to delete."
+              }
+            },
+            required: ["id"]
+          }
+        }
+      },
+      {
+        type: "function",
+        function: {
+          name: "getTasksForUser",
+          description: "Retrieve all tasks assigned to a specific user",
+          parameters: {
+            type: "object",
+            properties: {
+              user_id: {
+                type: "number",
+                description: "The user ID to retrieve tasks for."
+              }
+            },
+            required: ["user_id"]
+          }
+        }
+      },
+      {
+        type: "function",
+        function: {
+          name: "getTasksInSprint",
+          description: "Retrieve all tasks in a specific sprint with a given status",
+          parameters: {
+            type: "object",
+            properties: {
+              sprint_id: {
+                type: "number",
+                description: "The sprint ID to retrieve tasks from."
+              },
+              status: {
+                type: "string",
+                description: "The status of the tasks to retrieve."
+              }
+            },
+            required: ["sprint_id", "status"]
+          }
+        }
+      }
+    ]
+  });
+
+  let assistantId = assistant.id;
+  console.log("Created Assistant with Id: " + assistantId);
+
+  const thread = await openai.beta.threads.create({
+    messages: [
+      {
+        role: "user",
+        content: "What is the status of task 22?",
+      },
+    ],
+  });
+
+  let threadId = thread.id;
+  console.log("Created thread with Id: " + threadId);
+
+  let run = await openai.beta.threads.runs.createAndPoll(thread.id, {
+    assistant_id: assistantId,
+    additional_instructions: "Please address the user as Joel.",
   });
+
+  console.log("Run status: " + run.status);
+
+  if (run.status === "requires_action") {
+    await handleRequiresAction(run, thread.id);
+  } else {
+    console.log("Run completed without needing additional actions.");
+  }
 }
 
-main();
+async function handleRequiresAction(run: { required_action: { submit_tool_outputs: { tool_calls: any[]; }; }; id: any; }, threadId: any) {
```

Comment:
Is `run` needed in this function? It looks like you're accessing `run.required_action.submit_tool_outputs.tool_calls` -- might be better just to pass that as an argument instead? You're also potentially reassigning it in line 195.

If `run` is needed, you can create a type for the function:
```javascript
type T = { required_action: { submit_tool_outputs: { tool_calls: any[]; }; }; id: any; };
```

And use that type in the function name:
```javascript
async function handleRequiresActions(run: T, threadId: any)
```

Reply:
Nitpick: is there a type for `threadId`?

Reply:
Thanks for pointing this out, this review actually helped fixed most of my bugs in the program! I hadn't fully converted the previous code to this new Class properly, so there were little issues like that. Now most of them say `this.openai`, `this.thread`, `this.thread.id`, `this.run`, etc. so account for since these need to be accessed accross multiple methods in the class.

Reply:
> Nitpick: is there a type for `threadId`?

I need to look into the types for all of this APIs objects. So far this is the only info I've found regarding the objects, and it doesn't mention types. I may need to dive further into the docs.

![image](https://github.com/lookout-team/lookout/assets/117242786/e0e85479-bbce-4dcd-a03e-44bc400581e3)